### ISSUE TITLE
GWT preloader doesn't use percent-encoding

### ISSUE THREAD

**Issue by Frosty-J** [libgdx/libgdx#7356]
I noticed today a filename containing a semicolon fails to load on GWT as its path in `assets.txt` contains a literal `;` instead of `%3B`. This and other [reserved characters](https://en.wikipedia.org/wiki/Percent-encoding#Reserved_characters) should be encoded to avoid issues.

https://github.com/libgdx/libgdx/blob/97071719d19be261095122ec1d77af8e8bcdea84/backends/gdx-backends-gwt/src/com/badlogic/gdx/backends/gwt/preloader/PreloaderBundleGenerator.java#L125

It would also be an idea to transform the paths to lowercase, though the superDev server is case-sensitive even on Windows so any errors from that should be caught pretty quickly.

Sure, asset names shouldn't contain funny characters to begin with, but they do.

**Comment by venktre** [comment]
Can you assign this task to me?
It would be better if you could give me more guidance.

**Comment by Frosty-J** [comment]
No-one's rushing to do anything about this - it can informally be your issue. One possible solution could be:

```java
URLEncoder.encode("Hey; wh@t?!", "UTF-8").replace("+", "%20");
```
Returns: `"Hey%3B%20wh%40t%3F%21"`
